heat/core/io.py

```diff
@@ -406,14 +409,179 @@ def load(path, *args, **kwargs):
     else:
         raise ValueError('Unsupported file extension {}'.format(extension))
 
 
+def load_csv(path, header_lines=0, sep=';', dtype=types.float32, encoding='UTF-8', device=None, comm=MPI_WORLD):
+    """
+    Loads data from an CSV file. The data will be distributed along the 0 axis.
+
+    Parameters
+    ----------
+    path : str
+        Path to the CSV file to be read.
+    header_lines : int, optional
+        The number of columns at the beginning of the file that should not be considered as data.
+        default: 0.
+    sep : str, optional
+        The single char or string that separates the values in each row.
+        default: ';'
+    dtype : ht.dtype, optional
+        Data type of the resulting array;
+        default: ht.float32.
+    encoding : str, optional
+        The type of encoding which will be used to interpret the lines of the csv file as strings.
+        default: 'UTF-8'
+    device : None or str, optional
+        The device id on which to place the data, defaults to globally set default device.
+    comm : Communication, optional
+        The communication to use for the data distribution. defaults to MPI_COMM_WORLD.
+
+    Returns
+    -------
+    out : ht.DNDarray
+        Data read from the CSV file.
+
+    Raises
+    -------
+    TypeError
+        If any of the input parameters are not of correct type
+
+    Examples
+    --------
+    >>> import heat as ht
+    >>> a = ht.load_csv('data.csv')
+    >>> a.shape
+    [0/3] (150, 4)
+    [1/3] (150, 4)
+    [2/3] (150, 4)
+    [3/3] (150, 4)
+    >>> a.lshape
+    [0/3] (38, 4)
+    [1/3] (38, 4)
+    [2/3] (37, 4)
+    [3/3] (37, 4)
+    >>> b = ht.load_csv('data.csv', header_lines=10)
+    >>> b.shape
+    [0/3] (140, 4)
+    [1/3] (140, 4)
+    [2/3] (140, 4)
+    [3/3] (140, 4)
+    >>> b.lshape
+    [0/3] (35, 4)
+    [1/3] (35, 4)
+    [2/3] (35, 4)
+    [3/3] (35, 4)
+    """
+    if not isinstance(path, str):
+        raise TypeError('path must be str, not {}'.format(type(path)))
+    if not isinstance(sep, str):
+        raise TypeError('separator must be str, not {}'.format(type(sep)))
+    if not isinstance(header_lines, int):
+        raise TypeError('header_lines must int, not {}'.format(type(header_lines)))
+
+    # infer the type and communicator for the loaded array
+    dtype = types.canonical_heat_type(dtype)
+    # determine the comm and device the data will be placed on
+    device = devices.sanitize_device(device)
+    comm = sanitize_comm(comm)
+
+    file_size = os.stat(path).st_size
+    rank = comm.rank
+    size = comm.size
+    counts, displs, _ = comm.counts_displs_shape((file_size, 1), 0)
+
+    # Read a chunk of bytes and count the linebreaks
+    f = open(path, 'rb')
+    f.seek(displs[rank], 0)
+    line_starts = []
+    r = f.read(counts[rank])
+    for pos, l in enumerate(r):
+        if chr(l) == '\n':
+            line_starts.append(pos + 1)
+
+    if rank == 0:
+        line_starts = [0] + line_starts
```

Comment:
Do we need to store all line ending or rather only the number of lines locally and then research the buffer?

Reply:
This makes it more easy to read in the lines later. Do you think it costs to much memory to store it?

Reply:
Potentially. I am trying to determine right now, whether there will actually be so large CSV files that this will be an issue. For now we can leave it as is